### PR TITLE
add s-list__horizontal and docs

### PR DESCRIPTION
**Pull request by benjamin-hodgson**
Screenshot 2020-12-03 at 3 12 00 PM

### CODE REVIEW

File: lib/css/components/_stacks-lists.less [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+.s-list.s-list__horizontal {
+    list-style: none;
+    margin: 0;
+    padding: 0;
+
+    display: flex;
+    flex-direction: row;
+    flex-wrap: wrap;
+
+    > li:not(:last-child)::after {
```
Comment (benjamin-hodgson) [inline lib/css/components/_stacks-lists.less]: @aaronshekey Do you prefer `:not(:last-child)::after` or `:not(:first-child)::before` for the bullets? It'd affect text styling applied to the `li`s and also the wrapping behaviour when it goes to a second line

File: lib/css/components/_stacks-lists.less [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        content: '\2022';  // U+2022 BULLET
+        margin: 0 @su2;
```
Comment (benjamin-hodgson) [inline lib/css/components/_stacks-lists.less]: @aaronshekey I suppose we'll need a way to control the spacing?

**Comment by aaronshekey** [comment]
The organization / naming of this is a little weird. This is the first component we have that also has atomic classes of the same name. There's a `List` page in both the base classes and components 🤔 

Would calling this something like separators make sense? It's mildly weird that this component only has the horizontal modifier. Would we move all list styling over here? Or can this be a tiny component within our base styles. This one has got me thinking too much.

Additionally, on the core side of things I haven't quite found any examples of these separators. We've gotten rid of lots of `|` throughout the product, instead opting for simple white space and font color changes. Should we get with @PiperLawson to consider alternate treatments?
image

Sometimes the best component is no component.

File: lib/css/components/_stacks-lists.less [hunk 3]
```diff
@@ -21,8 +21,8 @@
     display: flex;
     flex-wrap: wrap;
 
-        content: '\2022';  // U+2022 BULLET
     > li:not(:last-child):after {
+        content: '•';
```
Comment (benjamin-hodgson) [inline lib/css/components/_stacks-lists.less:25]: I don't recommend using Unicode in code. This is liable to behave unpredictably depending on whatever default encoding the useragent has. Stick with the `\u` literal

**Review by aaronshekey — CHANGES_REQUESTED** [review]
As discussed in Slack, I'm taking the official stance that instead of horizontal lists with bulleted dividers, we should be using white space instead. This doesn't need to be a component, and we should shy away from this Talent-only aesthetic moving forward.

**Comment by aaronshekey** [comment]
Closing in favor of whitespace-only in our horizontal lists. 😞

**Pull request closed without merging**